fbs.field.inc

```diff
@@ -260,6 +276,16 @@ function fbs_field_metadata_get($entity, array $options, $name, $entity_type, &$
   return NULL;
 }
 
+/**
+ * Callback to set the value of a field.
+ */
+function fbs_field_metadata_set($entity, $name, $items, $langcode, $entity_type) {
+  if (!isset($items['value'])) {
+    $items = array('value' => $items);
```

Comment:
Existence of a certain key value is performed, however whole `$items` array is overriden.
Would suggest to assign default value to the actual key that is checked.